Code/cmake/Modules/generate_msvc_version.cmake

```diff
@@ -0,0 +1,107 @@
+include (CMakeParseArguments)
```

Comment:
I should add the stack overflow where I got these recipes

Reply:
yes please